### ISSUE TITLE
Tooltip for Invitation Lock

### ISSUE THREAD

**Issue by dmtrjsg** [Joystream/pioneer#3479]
## Scope

- Add tooltip to invitation lock

"Invitation lock can be spent on transaction fees and staking for proposals, voting and working groups applications. JOY tokens subject to this lock cannot be transferred to any other accounts. This lock is unrecoverable. NB: Transaction fees will first be taken from your transferable balance if it is positive."

Learn more about locks>: https://joystream.gitbook.io/testnet-workspace/system/accounts-and-staking"

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1201958687417145/1202809828964348) by [Unito](https://www.unito.io)

**Comment by toiletgranny** [comment]
Alright, @dmtrjsg, here's what I think we could do! https://www.figma.com/file/GlgN8uBRtvtMJtiOsdtDF7/Pioneer-Design?node-id=10813%3A421425 

Notes:
1. I've been experimenting with other places to show invitation locks across the "My Profile" page (due to its different nature), but, trying to devote a reasonable amount of time for these explorations, I couldn't come up with anything that would feel right. All my experiments would end up looking more confusing and complicated, rather than keeping the Invitation lock on the same list, together with all the other locks.
2. Instead, to emphasize this slightly different nature of the account lock, I made the "Restricted Usage" label more prominent than you suggested. I also applied this subtle line pattern in the background to communicate that this is more of a permanent state than other locks underneath, which, unlike the Invitation lock, will all eventually become claimable.

**Comment by dmtrjsg** [comment]
good thinking, if I say we can go with a green colour and "Restricted Usage Supported" to emphasise the positive connotation of usage, rather than restriction.  wdyt?

Re naming, Invitation lock name is best you recon?

**Comment by toiletgranny** [comment]
Yeah, @dmtrjsg, I think the green color works even better! Here, have a look and let me know if you agree so that I can update the other screens: https://www.figma.com/file/GlgN8uBRtvtMJtiOsdtDF7/Pioneer-Design?node-id=10821%3A414947

---

I missed the name part, sorry! Remembered that we wanted to tackle this in a separate task. Anyway, gave it some thought and here's my take on this.

Based on my current understanding of this, I think we need to be careful not to overly shift the name towards the "bonus", "gift" side. While that's true, you receive these tokens upon creating a membership for nothing (gift), unlike a typical donation, you'll need to repay it if your total balance goes below the initial invitation lock amount (debt, burden), correct? For this reason, all the names I would consider initially, no longer seem to be really appropriate, for example:

- Invitation bonus,
- Welcome bonus,
- Invitation grant,
- Welcome gift,
- Invitation gift,
- Starter gift,
- etc.

To avoid confusion amongst new members, I believe the name should emphasize the repayable nature of the grant, while communicating the good intentions behind it. Here's what I would think of:

- Starter pool,
- Starter credit,
- Welcome pool,
- Onboarding allowance,
- Base pool.

All in all, I think I would recommend going for something like that "Base pool" and adjusting the communication in the membership creation flow, eg. "We transferred your a one-time bonus of XXX JOY tokens for your Base pool.", or "We topped up your account with XXX JOY tokens for your Base pool to help you get started! Tokens in your Base pool are non-transferable and can only be spent on transaction fees and staking by this account."

Let me know if it makes sense. Once (or if) we agree on a new name, I'll also need to revisit Atlas designs to update the copy.
cc @bedeho

**Comment by bedeho** [comment]
>  you'll need to repay it if your total balance goes below the initial invitation lock amount (debt, burden), correct?

No, not really, perhaps you are getting the invite lock mixed up with bloat bonds, where the idea is that whenever you do an action, you have to put up a bit of $JOY which you later can recover when/if you reverse the action.

**Comment by toiletgranny** [comment]
Ok, sorry if I'm confusing things. Can I break down my current understanding of this and ask you to point out where am I mistaken? 

1. I create a new Joystream membership, and I receive 300 JOY tokens, all of which are under this invitation lock. And that's my entire account balance at the moment, I have nothing else. 
2. I'm a prolific Pioneer user, I engage in a bunch of activities for which I need to pay TX fees. All my tokens under invitation lock are now gone. My total account balance is 0.
3. Then I receive a transfer for 300 JOY tokens from you. I understand that all these tokens go out to refill the invitation lock balance automatically, so I end up where I started — 300 JOY tokens under invitation lock, 0 transferable JOY tokens.

**Comment by bedeho** [comment]
This is all correct, but not sure who `you` is, is it like a faucet or a friend or something?

I guess given that you understood it all, I possibly just misunderstood what you meant when you meant "repay". Its not that you are obligated to "pay someone" in order ot use the system, but if you are using the same account as before, then the lock persists, and so any new funds added will be impacted by the lock. You can of course change your account to one which does not have the old lock if you care about it.

**Comment by dmtrjsg** [comment]
@toiletgranny 

"Welcome gift" is good imho, perhaps even take it further and go with "New Membership Gift"

This is in-line with the new feature that existing members will be able to call `giftMembership` extrinsic, which credits destination acc and debits the acc specified by user, and thereby sponsor someone with new membership. Same lock will apply there. 

So let's go with "New Membership Gift" and also do the same for Atlas?

cc
@kdembler @bedeho

**Comment by bedeho** [comment]
I think you guys are overthinking this 100x  :D

No user who does not know about locks will be looking into this trying to understand the significance of each lock, and mapping that to what they can and cannot do. Each product, say like Atlas, needs to treat the presence of locks properly and explain to users what they can or cannot do at any given time, as we have aspired to do with the new balance treamtent. Exactly 0 users from Atlas will go from that product, confused by what is going on, and then try to debug it by interrogating lock list in Pioneer.

I would not call this a "gift", the purpose of the whole scheme is to allow people to not have to think about fees and tokens to get started using the product, its not to reward them, the ideal outcome is for them to not even be aware that invitation locks exist. I would just say it is the invitation lock, if needed say what it does in less than 10 words in a tooltip if that is truly needed, and move on.

**Comment by dmtrjsg** [comment]
>Overthinking 
yeh you are probably right.. happens sometimes by Fridays 😅 

ok then lets go with no change on Atlas and just a label "restricted usage allowed". In our defence, to my memory renaming Invitation lock was discussed sometime recently but let's not dwell on this for now 👍

**Comment by dmtrjsg** [comment]
Addressed in 3518

**Comment by ivanturlakov** [comment]
@traumschule  That's what needed to be added, did I get it right?

<img width="1052" alt="Снимок экрана 2022-11-04 в 14 29 56" src="https://user-images.githubusercontent.com/46903215/199973015-1707693a-1cf5-4ca6-8d0c-7acc68ee8f45.png">

Looks like it hasn't been added yet.

https://pioneer-2-joystream.vercel.app/#/profile

<img width="1175" alt="Снимок экрана 2022-11-04 в 14 27 33" src="https://user-images.githubusercontent.com/46903215/199972616-daa627dc-1f70-4d94-a77c-3ecbd51676ab.png">

https://dao-git-dev-joystream.vercel.app/#/profile

<img width="1177" alt="Снимок экрана 2022-11-04 в 14 29 01" src="https://user-images.githubusercontent.com/46903215/199972853-c3b2ba3e-56ef-4c47-8de4-ab7753ea4815.png">


https://dao-git-carthage-joystream.vercel.app/#/profile

<img width="1176" alt="Снимок экрана 2022-11-04 в 14 36 17" src="https://user-images.githubusercontent.com/46903215/199974119-adc5f2e4-5ed9-405c-ab8d-3ab0ae6b68ef.png">

**Comment by traumschule** [comment]
The modal was changed on `BuyMembershipSignModal` that's the signing confirmation creating a membership. I had to remove the `shouldInformAboutLock` to see it because the wallet is too rich. (the PR didn't change logic)

![sign-buy-membership](https://user-images.githubusercontent.com/31551045/200062577-925dfc44-4c05-46a8-be35-0337cb4e4819.png)

> That's what needed to be added, did I get it right?

It's supposed to look like this on My Accounts, i agree that tooltip is missing.

![inv-lock-tooltip](https://user-images.githubusercontent.com/31551045/200061107-fb6bb438-51bf-47f4-a0bb-e38e8104ccf3.png)
[Figma](https://www.figma.com/file/GlgN8uBRtvtMJtiOsdtDF7/Pioneer-Design?node-id=10821%3A414947)

**Comment by dmtrjsg** [comment]
@traumschule pls have a look

**Comment by traumschule** [comment]
Please try #3878: https://pioneer-2-git-fork-traumschule-fix-invitiation-6649eb-joystream.vercel.app

**Comment by polikosi** [comment]
tested on: https://dao-git-dev-joystream.vercel.app/#/settings
wss://34.230.5.182.nip.io/ws-rpc
![image](https://user-images.githubusercontent.com/80429191/209335504-4c38f3a4-929e-4a64-b7d8-92b455fabef8.png)

**Comment by traumschule** [comment]
Looks like it's in the wrong place. @mkbeefcake please take care of it. Thanks!

**Comment by thesan** [comment]
@traumschule the message here looks correct to me:
![](https://user-images.githubusercontent.com/80429191/209335504-4c38f3a4-929e-4a64-b7d8-92b455fabef8.png)

or am I missing something?

**Comment by traumschule** [comment]
@thesan yes tooltip message is correct and it's fine me with, just not according to design: should appear on "Invititation".
